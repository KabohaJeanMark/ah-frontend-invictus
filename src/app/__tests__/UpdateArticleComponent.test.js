import React from 'react';
import { shallow } from "enzyme";
import UpdateArticleComponent from "../components/UpdateArticle";
import toJson from 'enzyme-to-json';

describe('<UpdateArticleComponent/>', () => {
    let wrapper;
    const Props = {
        handleChange: jest.fn(),
        handleSubmit: jest.fn(),
        updateArticleErrors: { errors: {} },
        localState: {
            title: "",
            body: "",
            description: ""
        }
    };
    it('renders update article page', () => {
        wrapper = shallow(<UpdateArticleComponent {...Props} />);
    });
    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

