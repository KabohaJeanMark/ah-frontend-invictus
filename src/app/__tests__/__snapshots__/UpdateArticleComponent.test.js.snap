// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateArticleComponent/> matches snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-md-2"
    />
    <div
      className="col-md-8"
    >
      <div
        className="card article-card"
      >
        <center>
          <h5
            className="edit-mode"
          >
            Edit Mode
          </h5>
        </center>
        <form
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <textarea
              className="title-input"
              id="title"
              onChange={[MockFunction]}
              placeholder="Title"
              value=""
            />
            <p
              className="error-display"
            />
          </div>
          <div>
            <img
              alt=""
              className="card-image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="description-input"
              id="description"
              onChange={[MockFunction]}
              placeholder="Description"
              value=""
            />
            <p
              className="error-display"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="body-input"
              id="body"
              onChange={[MockFunction]}
              placeholder="Body"
              value=""
            />
            <p
              className="error-display"
            />
          </div>
          <button
            className="postarticle-btn"
          >
            Update
          </button>
        </form>
      </div>
    </div>
    <div
      className="col-md-2"
    />
  </div>
</Fragment>
`;
